syntax = "proto2";

package shardkv_messages;




message GetRequest{
	required int64 confignum=1;
	required string key=2; 
}

message GetResponse{
	required int64 flag=1;
	optional string leader_name=2; 
	optional string value=3;
}

message PutAppendRequest{
	required int64 requestid=1;	
	required int64 expirerequestid=2;
	required int64 confignum=3;
	required string key=4;
	required string value=5;
	required string op=6;
}

message PutAppendResponse{
	required int64 flag=1;
	optional string leader_name=2;
}


service ShardkvMessages {
    rpc Get (GetRequest) returns (GetResponse) {}
    rpc PutAppend (PutAppendRequest) returns (PutAppendResponse) {}
}




